addons/beehave/debug/tree_node.gd

```diff
@@ -154,37 +153,43 @@ func _check_for_conflicts(node: TreeNode) -> void:
 		for level in range(node.y + 1, min(sibling_contour.keys().max(), node_contour.keys().max()) + 1):
 			var distance: float = node_contour[level] - sibling_contour[level]
 			if distance + shift_value < min_distance:
-				shift_value = max(min_distance - distance, shift_value)
-
-		if shift_value > 0:
-			_center_nodes_between(node, sibling)
+				shift_value = min_distance - distance
+				shift_sibling = sibling
 
 		sibling = sibling.get_next_sibling()
 
 	if shift_value > 0:
 		node.x += shift_value
 		node.mod += shift_value
-		shift_value = 0
+		_center_nodes_between(shift_sibling, node)
 
 
 func _center_nodes_between(left_node: TreeNode, right_node: TreeNode) -> void:
 	var left_index := left_node.parent.children.find(left_node)
 	var right_index := left_node.parent.children.find(right_node)
 
-	var num_nodes_between: int = (left_index - right_index) - 1
+	var num_nodes_between: int = (right_index - left_index) - 1
 	if num_nodes_between > 0:
-		var distance_between_nodes: float = (left_node.x - right_node.x) / (num_nodes_between + 1)
-
-		var count: int = 1
+		# The extra distance that needs to be split into num_nodes_between + 1
+		# in order to find the new node spacing so that nodes are equally spaced
+		var distance_to_allocate: float = right_node.x - left_node.x - left_node.item.size.x
+		# Subtract sizes on nodes in between	
 		for i in range(left_index + 1, right_index):
-			var middle_node := left_node.parent.children[i]
-			var desire_x := right_node.x + (distance_between_nodes * count)
+			distance_to_allocate -= left_node.parent.children[i].item.size.x
+		# Divide space equally
+		var distance_between_nodes: float = distance_to_allocate / (num_nodes_between + 1)
```

Comment:
Split remaining distance after removing in between sibling sizes. This creates a uniform spacing between centered nodes, instead of centering just their `x` values